### PR TITLE
Set up for async actions

### PR DESCRIPTION
**Pull request by arinaldi**
Add promises and verify data returns from static API call.

### CODE REVIEW

File: src/actions/index.js [hunk 1]
```diff
@@ -1,60 +1,39 @@
-export const REQUEST_LOCATION = 'REQUEST_LOCATION';
 export const RECEIVE_LOCATION = 'RECEIVE_LOCATION';
-export const SELECT_LOCATION = 'SELECT_LOCATION';
-export const INVALIDATE_LOCATION = 'INVALIDATE_LOCATION';
 
-export function addLocation(location, air) {
+export function receiveLocation(location, json) {
+  //dispatching an action to the store
+  console.log(location, json);
```
Comment (ashleymichal) [inline src/actions/index.js]: 1. avoid committing console.log statements to your history.

File: src/actions/index.js [hunk 2]
```diff
@@ -1,60 +1,39 @@
...
-  location,
-  data: 'test', //json.data.children.map(child => child.data),
-  receivedAt: Date.now()
-});
-
+    type: 'RECEIVE_LOCATION',
+    date: json.datetime,
+    name: location,
+    aqi: json.breezometer_aqi,
+    desc: json.breezometer_description
```
Comment (ashleymichal) [inline src/actions/index.js]: go with full word `description` for your variable. we're not hurting for space, readability will be your friend in the future.

File: src/actions/index.js [hunk 3]
```diff
@@ -1,60 +1,39 @@
...
-const fetchPosts = location => dispatch => {
-  dispatch(requestLocation(location))
-  return fetch(`https://api.breezometer.com/baqi/?lat=40.7324296&lon=-73.9977264&key=fbbb89795db54612ad9598e2ce77b709&fields=breezometer_aqi,breezometer_color,breezometer_description,datetime`)
-    .then(response => response.json())
-    .then(json => dispatch(receiveLocation(location, json)))
+export const fetchLocation = location => {
+  return dispatch => {
+    //console.log(location);
+    //dispatch 'requesting'
+    //have parameter to fetch be Promise from my api module
```
Comment (ashleymichal) [inline src/actions/index.js]: remove unnecessary comments.

File: src/actions/index.js [hunk 4]
```diff
@@ -1,60 +1,39 @@
...
+    return fetch(`https://api.breezometer.com/baqi/?lat=40.7324296&lon=-73.9977264&key=fbbb89795db54612ad9598e2ce77b709&fields=breezometer_aqi,breezometer_color,breezometer_description,datetime`)
```
Comment (ashleymichal) [inline src/actions/index.js]: extract URLs and endpoint paths to constants

File: src/actions/index.js [hunk 5]
```diff
@@ -1,60 +1,39 @@
...
+      .then(response => response.json())
+      .then(json => {
+        //dispatch OK (gotten response)
+        //const json = response.json();
```
Comment (ashleymichal) [inline src/actions/index.js]: again.

File: src/actions/index.js [hunk 6]
```diff
@@ -1,60 +1,39 @@
...
+        dispatch(receiveLocation(location, json))
+        //console.log(response)
+
+        //return response
```
Comment (ashleymichal) [inline src/actions/index.js]: and here.. etc. If these are todos, mark them `// TODO: lorem ipsum...`

File: src/actions/index.js [hunk 7]
```diff
@@ -1,60 +1,39 @@
...
+      })
+      .catch(err => {})//dispatch error state)
```
Comment (ashleymichal) [inline src/actions/index.js]: avoid inline comments.

File: src/components/HistoryTable.js [hunk 8]
```diff
@@ -9,10 +9,12 @@ class HistoryTable extends React.Component {
   }
 
   renderLocation(location, i) {
-    return (
+     return (
```
Comment (ashleymichal) [inline src/components/HistoryTable.js]: intentional whitespace change? Tangentially, have we set up a linter for you?

File: src/containers/App.js [hunk 9]
```diff
@@ -1,15 +1,18 @@
 import React, { Component } from 'react';
-import InputForm from '../components/InputForm';
-import HistoryTable from '../components/HistoryTable';
 //import PropTypes from 'prop-types';
```
Comment (ashleymichal) [inline src/containers/App.js]: unused?

File: src/containers/App.js [hunk 10]
```diff
@@ -1,15 +1,18 @@
...
 import { connect } from 'react-redux';
+
+import { bindActionCreators } from 'redux';
+
 import * as actions from '../actions';
 
+import InputForm from '../components/InputForm';
+import HistoryTable from '../components/HistoryTable';
+
 class App extends Component {
```
Comment (ashleymichal) [inline src/containers/App.js]: this could be a pure function :-D

**Review by ashleymichal — CHANGES_REQUESTED** [review]

File: src/actions/index.js (hunk 1 above)
Comment (arinaldi) [reply src/actions/index.js]: Done.

File: src/actions/index.js (hunk 2 above)
Comment (arinaldi) [reply src/actions/index.js]: Done.

File: src/actions/index.js (hunk 3 above)
Comment (arinaldi) [reply src/actions/index.js]: Done.

File: src/actions/index.js (hunk 4 above)
Comment (arinaldi) [reply src/actions/index.js]: Done.

File: src/actions/index.js (hunk 5 above)
Comment (arinaldi) [reply src/actions/index.js]: Done.

File: src/actions/index.js (hunk 6 above)
Comment (arinaldi) [reply src/actions/index.js]: Done.

File: src/actions/index.js (hunk 7 above)
Comment (arinaldi) [reply src/actions/index.js]: Done.

File: src/components/HistoryTable.js (hunk 8 above)
Comment (arinaldi) [reply src/components/HistoryTable.js]: Done.

File: src/containers/App.js (hunk 9 above)
Comment (arinaldi) [reply src/containers/App.js]: Done.

File: src/reducers/index.js [hunk 11]
```diff
@@ -23,63 +21,3 @@ function locations(state = [], action) {
 const rootReducer = combineReducers({ locations });
```
Comment (ghaynes1292) [inline src/reducers/index.js]: You don't need to combine with a single reducer

File: src/containers/App.js (hunk 10 above)
Comment (arinaldi) [reply src/containers/App.js]: Done.

File: src/reducers/index.js (hunk 11 above)
Comment (arinaldi) [reply src/reducers/index.js]: Done.

**Review by ghaynes1292 — APPROVED** [review]

**Pull request merged**